parser/ParseStatement.hpp

```diff
@@ -276,29 +326,46 @@ class ParseStatementCreateIndex : public ParseStatement {
       inline_field_names->push_back("index_type");
       inline_field_values->push_back(index_type_->value());
 
-      if (attribute_name_list_.get() != nullptr) {
-        container_child_field_names->push_back("attribute_name_list");
+      if (attribute_list_ != nullptr) {
+        container_child_field_names->push_back("attribute_list");
         container_child_fields->emplace_back();
-        for (const ParseString& attribute_name : *attribute_name_list_) {
-          container_child_fields->back().push_back(&attribute_name);
+        for (const ParseAttribute &attribute : *attribute_list_) {
+          container_child_fields->back().push_back(&attribute);
         }
       }
 
-      if (index_property_list_.get() != nullptr) {
+      if (custom_properties_node_ != nullptr) {
         container_child_field_names->push_back("index_property_list");
         container_child_fields->emplace_back();
-        for (const ParseKeyValue& index_property : *index_property_list_) {
-          container_child_fields->back().push_back(&index_property);
-        }
+        container_child_fields->back().push_back(custom_properties_node_.get());
       }
     }
 
  private:
     std::unique_ptr<ParseString> index_name_;
     std::unique_ptr<ParseString> relation_name_;
-    std::unique_ptr<PtrList<ParseString> > attribute_name_list_;
+    std::unique_ptr<PtrList<ParseAttribute>> attribute_list_;
     std::unique_ptr<ParseString> index_type_;
-    std::unique_ptr<PtrList<ParseKeyValue> > index_property_list_;
+    std::unique_ptr<IndexProperties> index_properties_;
+    // Optional custom properties for the index can be specified during creation.
+    std::unique_ptr<ParseIndexProperties> custom_properties_node_;
```

Comment:
Do you need to change `custom_properties_node_` later? If not, mark it const by declaring `std::unique_ptr<const ParseIndexProperties> custom_properties_node_;`.